dbt/include/snowflake/macros/adapters.sql

```diff
@@ -150,21 +150,83 @@
 {% endmacro %}
 
 
-{% macro snowflake__list_relations_without_caching(schema_relation) %}
+{% macro snowflake__list_relations_without_caching(schema_relation, max_iter=10, max_results_per_iter=10000) %}
+
+  {%- set max_total_n = max_results_per_iter * max_iter -%}
+
   {%- set sql -%}
-    show terse objects in {{ schema_relation }}
+    show terse objects in {{ schema_relation }} limit {{ max_results_per_iter }}
   {%- endset -%}
 
   {%- set result = run_query(sql) -%}
-  {% set maximum = 10000 %}
-  {% if (result | length) >= maximum %}
-    {% set msg %}
-      Too many objects in schema  {{ schema_relation }}! dbt can only get
-      information about schemas with fewer than {{ maximum }} objects.
-    {% endset %}
-    {% do exceptions.raise_compiler_error(msg) %}
+  {%- set n = (result | length) -%}
+  {%- set counts_array = [n] -%}
```

Comment:
I'm feeling kind of dumb right now. What is the reason for this being a list? Why can't this just be a variable that we're adding two progressively as opposed to a list object that we're adding new results too. I feel like making this just a simple accumulator would simplify some of the control flow in this function